lib/HTTP/Server/Tiny.pm6

```diff
@@ -261,6 +261,7 @@ my class HTTP::Server::Tiny::Handler {
             if $result ~~ Promise {
                 return $result.result;
             }
+            warn "P6W app must return Promise";
```

Comment:
Yes. P6W app MUST return Promise when the app works in request-response protocol, but not in other protocols (like legacy psgi).